csv_permissions/permissions.py

```diff
@@ -182,168 +64,253 @@ def _parse_csv(
-                own = access_level.split(":", 1)[-1].strip()
-                if not own:
-                    message = (
-                        "No function name specified for 'own:'. Remove ':' or specify a" " function to call."
-                    )
-                    raise RuntimeError(message)
-
-                function_name = f"{model_name}_own_{own}"
-                access_level = "own"
-
-            elif access_level == "own":
-                function_name = f"{model_name}_own"
-                access_level = "own"
-
-            elif access_level.startswith("custom:"):
-                custom = access_level.split(":", 1)[-1].strip()
-                if not custom:
-                    raise RuntimeError("No custom function name specified.")
+    if resolve_permission_name is None:
+        resolve_permission_name = _default_resolve_permission_name
+    else:
+        resolve_permission_name = import_string(resolve_permission_name)
+
+    resolve_evaluators = tuple(
+        import_string(resolve_evaluator) if isinstance(resolve_evaluator, str) else resolve_evaluator
+        for resolve_evaluator
+        in resolve_evaluators
+    )
+
+    permission_is_global: Dict[PermName, bool] = {}
+
+    known_user_types: Set[UserType] = set()
+    known_perms: Set[PermName] = set()
+
+    permission_to_user_type_to_unresolved: Dict[PermName, Dict[UserType, UnresolvedEvaluator]] = {}
+
+    for file_path in file_paths:
+        file_permission_is_global, new_permission_to_user_type_to_unresolved, user_types = \
+            _parse_csv(file_path, resolve_permission_name)
+
+        # merge global list of known user types/permissions
+        known_user_types.update(set(user_types))
+        known_perms.update(set(file_permission_is_global.keys()))
+
+        # merge is_global settings
+        for permission, is_global in file_permission_is_global.items():
+            if permission in permission_is_global and permission_is_global[permission] != is_global:
+                raise ValueError(f"'Is Global' for {permission} in {file_path} is inconsistent with a previous CSV file")
+        permission_is_global.update(file_permission_is_global)
+
+        # merge unresolved permissions
+        for permission, new_user_type_to_unresolved in new_permission_to_user_type_to_unresolved.items():
+            if permission not in permission_to_user_type_to_unresolved:
+                permission_to_user_type_to_unresolved[permission] = {}
+            for user_type, new_unresolved in new_user_type_to_unresolved.items():
+                if user_type in permission_to_user_type_to_unresolved[permission]:
+                    if new_unresolved != permission_to_user_type_to_unresolved[permission][user_type]:
+                        raise ValueError(
+                            f"Permission {permission} for user type {user_type} in "
+                            f"{file_path} is inconsistent with a previous CSV file"
```

Comment:
is it possible for this and prev one to point out which previous csv file? (ie. a temp mapping to record which one gave what)

Reply:
(Please don't resolve this comment yet, leaving as a placeholder)

TODO: could add a `source` attribute to `UnresolvedEvaluator`

Reply:
It now reports a previous file that is inconsistent with the current one (also for the is_global check a little earlier)